src/platform/plugins/shared/chart_expressions/expression_partition_vis/public/utils/layers/get_layers.test.ts

```diff
@@ -119,56 +137,72 @@ describe('getLayers', () => {
   });
 
   it('should handle empty slices with default label', () => {
-    const visData = createMockVisData();
-    const visDataWithNullValues = {
-      ...visData,
-      rows: [
+    const visDataWithNullValues: Datatable = {
+      type: 'datatable',
+      columns: [
         {
-          'col-0-2': 'Null Airways',
-          'col-1-1': null,
-          'col-2-3': null,
-          'col-3-1': null,
+          id: 'category',
+          name: 'Category',
+          meta: {
+            type: 'string',
+            params: {
+              id: 'terms',
+              params: {
+                id: 'string',
+                otherBucketLabel: 'Passed other label',
+                missingBucket: true,
+                missingBucketLabel: 'Passed missing label',
+              },
+            },
+          },
+        },
+        {
+          id: 'value',
+          name: 'Value',
+          meta: { type: 'number', params: { id: 'number' } },
         },
       ],
+      rows: [
+        { category: '', value: 10 },
+        { category: null, value: 10 },
+        { category: MISSING_TOKEN, value: 10 },
+        { category: '__other__', value: 10 },
+      ],
     };
 
-    const columns: BucketColumns[] = [
-      {
-        id: 'col-0-0',
-        name: 'Normal column',
-        meta: { type: 'murmur3' },
-      },
-      {
-        id: 'col-0-0',
-        name: 'multi-metric column',
-        meta: {
-          type: 'number',
-          sourceParams: {
-            consolidatedMetricsColumn: true,
-          },
-        },
-      },
-    ];
     const visParams = createMockPieParams();
+    const formatters = generateFormatters(visDataWithNullValues, fieldFormatsRegistry.deserialize);
     const layers = getLayers(
       ChartTypes.PIE,
-      columns,
+      [visDataWithNullValues.columns[0]], // only the bucket column
       visParams,
       visDataWithNullValues,
       {},
       [],
       getPaletteRegistry(),
       palettes,
-      {},
-      fieldFormatsMock,
+      formatters,
+      fieldFormatsRegistry,
       false,
       false
     );
-
     for (const layer of layers) {
-      expect(layer.groupByRollup(visDataWithNullValues.rows[0], 0)).toEqual('(empty)');
-      expect(layer.showAccessor?.(visDataWithNullValues.rows[0]['col-0-2'])).toEqual(true);
-      expect(layer.nodeLabel?.(visDataWithNullValues.rows[0]['col-0-2'])).toEqual('Null Airways');
+      expect(layer.showAccessor?.(visDataWithNullValues.rows[0].category)).toEqual(true);
+      expect(layer.nodeLabel?.(visDataWithNullValues.rows[0].category)).toEqual(EMPTY_LABEL);
```

Comment:
Nit: There's pros/cons about asserting against the defined const and not the raw string. `toEqual(EMPTY_LABEL);` is easier to maintain. but `toEqual('(blank)')` makes the expected outcome more obvious. if `EMPTY_LABEL` would ever change, the assertions need updating, but it would also make it more obvious in git diffs which updates are affected. using the raw string in the assertions also would protect against accidental changes/typos in the const.

Reply:
you are right, I can bring this back in a different PR